src/valkey-cli.c

```diff
@@ -2427,6 +2405,25 @@ static int cliSendCommand(int argc, char **argv, long repeat) {
                             config.pubsub_mode = 1;
                             cliRefreshPrompt();
                         }
+
+                        /* Handle pubsub mode */
+                        if (config.last_reply->elements >= 3) {
```

Comment:
This code now is now inside an `if (num_expected_pubsub_push > 0 ...) { ... }` block.

For a command like UNSUBSCRIBE without channels specified, num_expected_pubsub_push is 1 here, so this code doesn't run if UNSUBSCRIBE returns multiple messages. UNSUBSCRIBE without channels actually unsubscribe all, so it can return multiple replies and the last count can be `0`.

So we need to move this code outside of `if (num_expected_pubsub_push > 0 ...) { ... }`. Put it before that `if`?